### PR TITLE
Switch to context instead of google/net/context. Update Find() params.

### PR DESCRIPTION
**Pull request by ultimateboy**
Thanks so much for this!

I did have to make a few modifications to get things to work initially. Here's a PR with those changes.

### CODE REVIEW

**Comment by ajcrowe** [comment]
Hi @ultimateboy 

Thanks for the update to 1.7's context, I see rest-layer has made this change in the last month so this should not conflict now. The page to offset change as well I had not seen that, so thanks!

I used the underscores to mirror the structure of the [mongo](https://github.com/rs/rest-layer-mongo/blob/master/mongo.go) storage library, it may be that this is a mongo convention however it should avoid conflicts, what is the rationale for changing it? Maybe having the normal name and skipping makes more sense.

The schema.go was simple me playing around with the idea of being able to implement custom storage fields but this is not currently possible, I'm talking to about this [here](https://github.com/rs/rest-layer/issues/72) so if you could remove those changes for now.

Otherwise I will merge this in.

Thanks!

File: schema.go [hunk 1]
```diff
@@ -9,21 +9,22 @@ type Fields map[string]Field
 // Wrap common schema fields from rest-layer
 var (
 	IDField = Field{
-		IDField,
+		Field:   schema.IDField,
+		NoIndex: false,
```
Comment (ajcrowe) [inline schema.go]: This is the default so isn't needed

File: schema.go [hunk 2]
```diff
@@ -9,21 +9,22 @@ type Fields map[string]Field
...
 	}
 
 	CreatedField = Field{
-		schema.CreatedField,
+		Field:   schema.CreatedField,
```
Comment (ajcrowe) [inline schema.go]: I embedded this directly so it doesn't have to be referenced by .Field

File: datastore.go [hunk 3]
```diff
@@ -221,18 +225,14 @@ func (d *Handler) Clear(ctx context.Context, lookup *resource.Lookup) (int, erro
 }
 
 // Find entities matching the provided lookup from the Datastore
-func (d *Handler) Find(ctx context.Context, lookup *resource.Lookup, page, perPage int) (*resource.ItemList, error) {
+func (d *Handler) Find(ctx context.Context, lookup *resource.Lookup, offset int, limit int) (*resource.ItemList, error) {
```
Comment (ajcrowe) [inline datastore.go]: This looks good, however lets just change it to `offset, limit int`

File: schema.go (hunk 2 above)
Comment (ultimateboy) [inline schema.go]: I get a `mixture of field:value and value initializers` error without this change.

File: schema.go (hunk 1 above)
Comment (ultimateboy) [inline schema.go]: Indeed. I added for clarity, but I'll remove.

File: datastore.go (hunk 3 above)
Comment (ultimateboy) [inline datastore.go]: Will update.

File: schema.go (hunk 2 above)
Comment (ajcrowe) [inline schema.go]: If you just remove this file completely, I've stuck it in a feature branch for now.

Comment (ultimateboy) [inline schema.go]: Done.

**Comment by ajcrowe** [comment]
Great thanks @ultimateboy

**Pull request merged**